src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8131,6 +8131,28 @@ public static string GetTarget(PSObject instance)
             return null;
         }
 
+        /// <summary>
+        /// Gets the target for a given file or directory, resolving symbolic links.
+        /// </summary>
+        /// <param name="instance">The FileInfo or DirectoryInfo type.</param>
+        /// <returns>The file path the instance points to.</returns>
+        public static FileSystemInfo ResolvedTarget(PSObject instance)
```

Comment:
The issue is asking for a full path (return string result), but you made it return `FileInfo` or `DirectoryInfo` instead, why not returning string as requested in the issue?
The existing properties `Target` and `LinkTarget` both return a string, so I think it's better to do the same for `ResolvedTarget` property.

Reply:
What is a negative effects from the throw? Should we really mask the exceptions?

Reply:
While do we need to limit the method? PowerShell does FileInfo.ToString() for us.

Reply:
We should be consistent with `Target` and `LinkTarget`, because the name `ResolvedTarget` indicates the same parameter type.

Reply:
> We should be consistent

Why? .Net object is more useful than string. And PowerShell will convert to string itself which it does in many scenarios.

Reply:
Well, if there is disagreement on the parameter type, then let's have it reviewed by the committee.

Reply:
Since you mentioned PSPath... :-) Last days I am thinking about FileSystem Provider V2 and related issues although it is more correct to say the problems are more related to _path conversions_. And now I'm more and more inclined to think that PSPath should be an object and not just a string, and the other ETS properties could be based on that, and it would great to inject this in PSObject itself. 
This is much more in the spirit of PowerShell, opens the door to fixing many long-standing bugs (by experimental feature(s)), improving engine and provider performance, and making cross-provider operations easier to implement.